server/api/src/rsDataObjRepl.cpp

```diff
@@ -728,31 +734,100 @@ namespace
         auto destination_cond_input = irods::experimental::make_key_value_proxy(destination_inp.condInput);
         auto destination_obj = resolve_hierarchy_and_get_data_object_info(_comm, destination_inp, irods::CREATE_OPERATION);
 
+        const auto* destination_hierarchy = destination_cond_input.at(DEST_RESC_HIER_STR_KW).value().data();
```

Comment:
Is the type of `.value()` a std::string or std::string_view?

If it's a std::string, prefer `.c_str()` instead of `.data()`.

Reply:
`value()` is a `std::string_view`

Reply:
See https://github.com/irods/irods/pull/7490#discussion_r1484421460